xbmc/application/Application.cpp

```diff
@@ -288,27 +288,20 @@ void CApplication::HandlePortEvents()
             settings->SetInt(CSettings::SETTING_WINDOW_HEIGHT, newEvent.resize.h);
             settings->Save();
           }
-#ifdef TARGET_WINDOWS
-          else
-          {
-            // this may occurs when OS tries to resize application window
-            //CDisplaySettings::GetInstance().SetCurrentResolution(RES_DESKTOP, true);
-            //auto& gfxContext = CServiceBroker::GetWinSystem()->GetGfxContext();
-            //gfxContext.SetVideoResolution(gfxContext.GetVideoResolution(), true);
-            // try to resize window back to it's full screen size
-            //! TODO: DX windowing should emit XBMC_FULLSCREEN_UPDATE instead with the proper dimensions
-            //! and position to avoid the ifdef in common code
-            auto& res_info = CDisplaySettings::GetInstance().GetResolutionInfo(RES_DESKTOP);
-            CServiceBroker::GetWinSystem()->ResizeWindow(res_info.iScreenWidth, res_info.iScreenHeight, 0, 0);
-          }
-#endif
         }
         break;
       case XBMC_FULLSCREEN_UPDATE:
       {
         if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_fullScreen)
         {
-          CServiceBroker::GetWinSystem()->ResizeWindow(newEvent.resize.w, newEvent.resize.h, 0, 0);
+          const int heigh = CServiceBroker::GetWinSystem()->GetHeight();
```

Comment:
This feels hacky, probably it's better as a new method on `CWindowSystemBase` which implementations are free to define - like `ForceFullscreen()` or `ResetFullScreen`